jerry-core/ecma/operations/ecma-exceptions.c

```diff
@@ -312,26 +314,14 @@ ecma_new_aggregate_error (ecma_value_t error_list_val, /**< errors list */
  *         if it is not an Error object then JERRY_ERROR_NONE will be returned
  */
 jerry_error_t
-ecma_get_error_type (ecma_object_t *error_object) /**< possible error object */
+ecma_get_error_type (ecma_object_t *error_object_p) /**< possible error object */
 {
-  if (error_object->u2.prototype_cp == JMEM_CP_NULL || ECMA_OBJECT_IS_PROXY (error_object))
+  if (!ecma_object_class_is (error_object_p, ECMA_OBJECT_CLASS_ERROR))
   {
     return JERRY_ERROR_NONE;
   }
 
-  ecma_object_t *prototype_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, error_object->u2.prototype_cp);
-
-  uint8_t builtin_id = ecma_get_object_builtin_id (prototype_p);
-
-  for (uint8_t idx = 0; idx < sizeof (ecma_error_mappings) / sizeof (ecma_error_mappings[0]); idx++)
-  {
-    if (ecma_error_mappings[idx].error_prototype_id == builtin_id)
-    {
-      return ecma_error_mappings[idx].error_type;
-    }
-  }
-
-  return JERRY_ERROR_NONE;
```

Comment:
AFAIK: After this then we would not need the `ecma_error_mappings` array at all (and related type).